Update sl-jdbc.md (#3953)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-cloud-apis/sl-jdbc.md

</TabItem>

<TabItem value="queryablegranularitiesformetrics" label="Fetch queryable primary time granularities for metrics">

> Use this query to fetch queryable granularities for a list of metrics. This API request allows you to only show the time granularities that make sense for the primary time dimension of the metrics (e.g., `metric_time`), but if you want queryable granularities for other time dimensions, you can use the `dimensions()` call, and find the column queryable_granularities.

Reviewer: ```suggestion
Use this query to fetch queryable granularities for a list of metrics. This API request allows you to only show the time granularities that make sense for the primary time dimension of the metrics (such as `metric_time`), but if you want queryable granularities for other time dimensions, you can use the `dimensions()` call, and find the column queryable_granularities.
```

### Query with where filters

> Where filters in API allow for a filter list or string. We recommend using the filter list for production application as this format will realize all benefits from predicate pushdown where possible. 

Reviewer: @rpourzand , what does predicate mean in this sentence?

```suggestion
In the API, you can use where filters either as a filter list or string. For production applications, it's better to use a filter list. This format helps you take advantage of all the benefits from the predicate pushdown where possible.
```

Author: So a predicate is just an expression to determine what rows in a database apply to a particular query. For example, if I filter in a WHERE clause on a particular dimension value, the database is basically going to search to determine what values in the DB apply to this query. If this can be pushed down to the database (predicate pushdown)), it's much faster.

Author: We could link out some generic docs on predicate pushdown.

Reviewer: Predicate Pushdown points to the where or filter clause which affects the number of rows returned. It basically relates to which rows will be filtered, not which columns. For this reason, while applying the filter on a nested column as ‘library.books’ to merely return records with the values that are not null, the predicate pushdown function will make parquet read blocks that contain no null values for the specified column.

> Where Filters have a few objects that can be used:

Reviewer: ```suggestion
Where Filters have a few objects that you can use:
```

- `Dimension()` - This is used for any categorical or time dimensions. If used for a time dimension, granularity is required -  `Dimension('metric_time').grain('week')` or `Dimension('customer__country')`

- `Entity()` - used for entities like primary and foreign keys - `Entity('order_id')`

> Note: If you prefer more strongly typed `where` clause, you can optionally use `TimeDimension` to separate out categorical dimensions from time ones. The `TimeDimesion` input takes the time dimension name and also requires granularity - an example is `TimeDimension('metric_time', 'MONTH')`.

Reviewer: ```suggestion
Note: If you prefer a more strongly typed `where` clause, you can optionally use the `TimeDimension` feature. This helps separate out categorical dimensions from time-related ones. The `TimeDimesion` input takes the time dimension name and also requires granularity, like this: `TimeDimension('metric_time', 'MONTH')`.
```

> Use the following example to query using a `where` filter using the string format:

Reviewer: ```suggestion
Use the following example to query using a `where` filter with the string format:
```
